### PR TITLE
add BLAS Level 1 `srotg` routine

### LINKED ISSUE #280: RFC: add BLAS Level 1 srotg routine
## Checklist

> Please ensure the following tasks are completed before submitting a feature request.

-   [X] Read and understood the [Code of Conduct][code-of-conduct].
-   [X] Searched for existing issues and pull requests.
-   [X] The issue name begins with `RFC:`.

## Description

> Description of the feature request.

This RFC proposes to add a routine to construct a Givens plane rotation as defined in [BLAS Level 1][blas-level-1] routines.  Specifically adding `@stdlib/blas/base/srotg` is proposed.

## Related Issues

> Does this feature request have any related issues?

No.

## Questions

> Any questions for reviewers?

No.

## Other

> Any other information relevant to this feature request? This may include screenshots, references, sample output, and/or implementation notes.

No.

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md

[blas-level-1]: http://www.netlib.org/blas/#_level_1

### PR DESCRIPTION
**Pull request by Pranavchiku**
Fixes #280.

Adding first js files, as I am familiar to convert js to C implementation.

## Checklist

- [x] Add readme.md
- [ ] docs
- [x] src
- [x] include/stdlib/blas/base
- [x] include.gypi
- [x] binding.gyp
- [x] manifest.json
- [x] package.json
- [ ] lib
- [ ] examples
- [ ] benchmark
- [ ] test 
* * *

@kgryte

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/base/srotg/lib/srotg.assign.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var abs = require( '@stdlib/math/base/special/abs' );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/base/srotg/lib/srotg.assign.js:23]: @kgryte if you can have a look at it, that might be of great help. I am not sure if it is correct implementation,

File: lib/node_modules/@stdlib/blas/base/srotg/lib/srotg.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var Float32Array = require( '@stdlib/array/float32' );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/base/srotg/lib/srotg.js:23]: And this one also.

Comment (kgryte) [reply lib/node_modules/@stdlib/blas/base/srotg/lib/srotg.js:23]: This one looks good.

**Comment by Pranavchiku** [comment]
Closing with reference to #806.

**Comment by kgryte** [comment]
Why close this one? Can we not just add the JS implementation here?

**Comment by Pranavchiku** [comment]
Yeah, we can, but I thought of doing a fresh implementation, as it was a bit messy. Pardon me for that.

**Pull request closed without merging**